Makefile

```diff
@@ -217,6 +218,23 @@ test-integration: build ## run the integration tests
 	$(DOCKER_RUN_DOCKER) hack/make.sh dynbinary test-integration
 endif
 
+test-buildkit: build
+	# Download buildkit
+	# TODO: change this to moby version of the project after
+	# https://github.com/moby/buildkit/pull/1164 is merged.
+	curl -fsSL -o "/tmp/buildkit.tar.gz" https://github.com/SamWhited/buildkit/archive/${BUILDKIT_COMMIT}.tar.gz
+	rm -rf ../buildkit/
+	tar -xzf "/tmp/buildkit.tar.gz"
+	mv buildkit-${BUILDKIT_COMMIT}/ ../buildkit/
+	rm -rf ../buildkit/vendor/github.com/docker/docker
+	cp -r "$(CURDIR)" ../buildkit/vendor/github.com/docker/docker
+
+	$(DOCKER_RUN_DOCKER) hack/make.sh binary-daemon binary
+	cd ../buildkit/ && \
+		TEST_DOCKERD=1 \
+		EXTRA_INTEGRATION_DOCKER_FLAGS="-v $(CURDIR)/bundles/binary-daemon/dockerd:/bin/dockerd:ro" \
+		./hack/test integration
```

Comment:
You are testing the docker daemon that's provided by Jenkins, not the docker daemon in the codebase. If you look at how `test-docker-py` works as an example, that runs inside the container: `$(DOCKER_RUN_DOCKER) hack/make.sh dynbinary test-docker-py` and buildkit should do the same. So you do need a `hack/make/test-buildkit` in moby that will call into buildkit's `/hack/test integration` from within the container.

Reply:
The reason this ended up being changed was because the buildkit integration tests also try to launch containers and the docker-in-docker stuff was likely to be a source of problems and confusion, although I don't remember the specific issues I was running into. I'll go reread the conversation with @thaJeztah and play around with reverting it back; thanks!